python/ccxt/pro/base/client.py

```diff
@@ -11,6 +11,7 @@ class Client(object):
     url = None
     ws = None
     futures = {}
+    options = {}  # ws-specific options
```

Comment:
@pcriadoperez ⬆️  this won't work because "options" does not exist inside `config`

`config` is basically the  ws options dict, and will assign its values dynamically,  so here we should define the specific option we want to look for, in this case, `headers = {}` 



![image](https://user-images.githubusercontent.com/43336371/209541315-57d03950-4124-4782-99bf-6fc93dd04859.png)

Reply:
"options" does exist inside config.

Here is the print of config:
```
{'keepAlive': 30000, 'ping': None, 'maxPingPongMisses': 2.0, 'log': <bound method Exchange.log of ccxt.async_support.bitopro()>, 'verbose': False, 'throttle': <ccxt.async_support.base.throttler.Throttler object at 0x7fdc3709f3a0>, 'asyncio_loop': <_UnixSelectorEventLoop running=True closed=False debug=False>, 'options': {'headers': {}}}
```

I tested with binance (doesn't require headers) and gemini (does require headers) and did work.

Reply:
@pcriadoperez weird, check this out 
![image](https://user-images.githubusercontent.com/43336371/209574914-75dea598-8d4f-4030-8540-5c2d7d104c2b.png)

Reply:
@carlosmiei  what exchange you are using? Does it have anything under `this.options['ws']['options']['headers']` ?

Reply:
@pcriadoperez I used Huobi, it has stuff inside `this.options['ws']['options']`

Reply:
@carlosmiei, so huobi I see it has `this.options['ws']` but not `this.options['ws']['options']` which is what's captured.

Reply:
> @carlosmiei, so huobi I see it has `this.options['ws']` but not `this.options['ws']['options']` which is what's captured.

@pcriadoperez exactly, what's the point of having another `options` ?

Reply:
@carlosmiei I agree it's a bit redundant.. but it's implemented like that in hte JS and php client, so thought it was better not to change and keep the same in python

Reply:
@pcriadoperez are you talking about this? 
![image](https://user-images.githubusercontent.com/43336371/209642462-864f9890-223d-43b8-96e1-51dc906627fe.png)
If so, seems a bit different to me

Reply:
@pcriadoperez any update here? 🙂

Reply:
hey @carlosmiei, so there it is also stored in `this.options['ws']['options']['headers']` As it grabs the `this.options['ws'] in line 16 of the screenshot and then extends it in line 6. (By the way why does you line count in the picture go backwards 🤣 ? )
Php also works in the same way:
![image](https://user-images.githubusercontent.com/12142844/210103403-98930ef6-f72a-4dcd-9ce7-0c639cc5b3f4.png)
![image](https://user-images.githubusercontent.com/12142844/210103451-7f34b198-b114-454d-92c5-58cfd859ca79.png)

I agree it's a bit confusing I believe it was implemente this way because we pass `this.options['ws']` as config to the 'ws' library constructor, and then the headers are grabbed from `options.headers`  inside the lirary (https://github.com/websockets/ws/blob/master/doc/ws.md#new-websocketaddress-protocols-options , https://nodejs.org/api/http.html#http_http_request_options_callback)

Bitopro and the branch of gemini-ws are good to test as they are the both we have that use headers.

On another note you made me realize we have another bug, as we are overwritting the 'agent' if we pass in headers or any other options. I'll open a PR for it.

Reply:
@pcriadoperez 
> By the way why does you line count in the picture go backwards 🤣 

I live in the upside world  🤣 
(jk using relative line numbers)
![image](https://user-images.githubusercontent.com/43336371/210137329-63e440c0-40f1-4b34-b0b7-f6ba6f027ffc.png)